src/GraphQL.php

```diff
@@ -249,18 +252,82 @@ public static function execute(
         array $variableValues = [],
         $operationName = null,
         callable $fieldResolver = null,
-        $errorHandler = null
+        ?ErrorHandlerInterface $errorHandler = null
     ): ExecutionResult {
         /** @var ExecutionInterface $execution */
         $execution = static::make(ExecutionInterface::class);
 
-        if (null !== $errorHandler) {
-            $errorHandler = $errorHandler instanceof ErrorHandlerInterface
-                ? $errorHandler
-                : new ErrorHandler($errorHandler);
+        return $execution->execute(
+            $schema,
+            $document,
+            $rootValue,
+            $contextValue,
+            $variableValues,
+            $operationName,
+            $fieldResolver,
+            $errorHandler
+        );
+    }
+
+    /**
+     * @param Schema                     $schema
+     * @param string                     $source
+     * @param null                       $rootValue
+     * @param null                       $contextValue
+     * @param array                      $variableValues
+     * @param null|string                $operationName
+     * @param callable|null              $fieldResolver
+     * @param ErrorHandlerInterface|null $errorHandler
+     * @return ExecutionResult
+     * @throws InvariantException
+     * @throws SyntaxErrorException
+     */
+    public static function process(
```

Comment:
Why move this to a static method?

Reply:
This wasn't a part of the `GraphQL` class at all before, only the public API.